x-pack/plugins/security_solution/public/management/pages/event_filters/store/action.ts

```diff
@@ -22,6 +22,18 @@ export type EventFiltersListPageDataExistsChanged = Action<'eventFiltersListPage
   payload: EventFiltersListPageState['listPage']['dataExist'];
 };
 
+export type EventFilterForDeletion = Action<'eventFilterForDeletion'> & {
+  payload: Pick<ExceptionListItemSchema, 'id' | 'name'>;
```

Comment:
Why do you Pick the `id` and `name` if you are storing the whole item in the store? Probably I'm missing something but if you want to store just the name and the id I think you should Pick this values on the dispatch right? (at `x-pack/plugins/security_solution/public/management/pages/event_filters/view/event_filters_list_page.tsx`)

Reply:
I wanted to store the entire event filter record, but was getting Type errors around the fact that record is Immutable _(I think it was the comments array that was causing it)_ and the action here was defining it as non-immutable (as it should be, IMO). so instead of casting, I opted to define  it as only requiring `id` and `name`.

Agree this is a bit miss-leading. I can change it to just store the entire record. Maybe I can use `MyabeImmutable<>` generic here.

Reply:
Fixed. Will accept full exception item.